packages/stark-ui/src/modules/input-mask-directives/directives/number-mask.directive.ts

```diff
@@ -39,89 +40,80 @@ export const STARK_NUMBER_MASK_VALUE_ACCESSOR: Provider = {
-		allowNegative: true,
-		allowLeadingZeroes: false
-	};
-
-	/**
-	 * Class constructor
-	 * @param _renderer - Angular `Renderer2` wrapper for DOM manipulations.
-	 * @param _elementRef - Reference to the DOM element where this directive is applied to.
-	 * @param _compositionMode - Injected token to control if form directives buffer IME input until the "compositionend" event occurs.
-	 */
 	public constructor(
 		_renderer: Renderer2,
 		_elementRef: ElementRef,
+		_factory: IMaskFactory,
+		@Inject(PLATFORM_ID) _platformId: string,
 		@Optional() @Inject(COMPOSITION_BUFFER_MODE) _compositionMode: boolean
 	) {
-		super(_renderer, _elementRef, _compositionMode);
-		this.elementRef = _elementRef;
+		super(_renderer, _elementRef, _factory, _platformId, _compositionMode);
 	}
 
-	/**
-	 * Component lifecycle hook
-	 */
-	public override ngOnChanges(changes: SimpleChanges): void {
-		this.textMaskConfig = this.normalizeMaskConfig(this.maskConfig);
+	public override normalizeMaskConfig(maskConfig: string | StarkNumberMaskConfig, defaultMask: StarkNumberMaskConfig): any {
+		if (!maskConfig) {
+			return undefined;
+		}
+		const mask: StarkNumberMaskConfig = this.mergeMaskConfig(maskConfig, defaultMask);
 
-		super.ngOnChanges(changes);
+		const numberMask: IMask.MaskedNumberOptions = {
+			mask: Number,
+			scale: mask.allowDecimal ? mask.decimalLimit : 0,
+			signed: mask.allowNegative,
+			thousandsSeparator: mask.includeThousandsSeparator ? mask.thousandsSeparatorSymbol : "",
+			padFractionalZeros: mask.allowLeadingZeroes,
+			normalizeZeros: mask.requireDecimal,
+			radix: mask.decimalSymbol,
+			max: mask.integerLimit ? Math.pow(10, mask.integerLimit) - 1 : undefined,
+			min: mask.integerLimit && mask.allowNegative ? Math.pow(10, mask.integerLimit) * -1 + 1 : undefined
+		};
 
-		// TODO: temporary workaround to update the model when the maskConfig changes since this is not yet implemented in text-mask and still being discussed
-		// see: https://github.com/text-mask/text-mask/issues/657
-		if (changes["maskConfig"] && !changes["maskConfig"].isFirstChange() && this.textMaskConfig.mask !== false) {
-			// trigger a dummy "input" event in the input to trigger the changes in the model (only if the mask was not disabled!)
-			const ev: Event = document.createEvent("Event");
-			ev.initEvent("input", true, true);
-			(<HTMLInputElement>this.elementRef.nativeElement).dispatchEvent(ev);
+		if (!!mask.suffix || !!mask.prefix) {
+			return {
+				mask: (mask.prefix ? mask.prefix : "") + " block " + (mask.suffix ? mask.suffix : ""),
+				blocks: {
+					block: numberMask
+				}
+			};
 		}
+		return numberMask;
 	}
 
-	/**
-	 * Create a valid configuration to be passed to the MaskedInputDirective
-	 * @param maskConfig - The provided configuration via the directive's input
-	 */
-	public normalizeMaskConfig(maskConfig: StarkNumberMaskConfig): Ng2TextMaskConfig {
-		if (typeof maskConfig === "undefined") {
-			return { mask: false }; // remove the mask
```

Comment:
What if the directive receives `false`, do you disable the mask ?

Reply:
This is managed on the base class like for email